packages/@react-aria/virtualizer/src/Virtualizer.tsx

```diff
@@ -185,25 +185,16 @@ export function useVirtualizer<T extends object, V, W>(props: VirtualizerOptions
     isFocusWithin.current = ref.current.contains(e.relatedTarget as Element);
   }, [ref]);
 
-  // When the focused item is scrolled out of view and is removed from the DOM,
-  // move focus to the collection view as a whole if focus was within before.
-  let focusedView = virtualizer.getView(focusedKey);
-  useEffect(() => {
-    if (focusedKey && !focusedView && isFocusWithin.current && document.activeElement !== ref.current) {
-      focusWithoutScrolling(ref.current);
-    }
-  });
```

Comment:
Don't we still need something to move focus to the collection if the focused row is deleted rather than just scrolled out of view? This story loses focus to the body when the last row is deleted: https://reactspectrum.blob.core.windows.net/reactspectrum/1413f7c3690cb18a6655328eaf20221de79ca7f2/storybook/iframe.html?scrolling=false&strict=false&args=&providerSwitcher-locale=&providerSwitcher-theme=&providerSwitcher-scale=&providerSwitcher-express=false&id=tableview--inline-delete-buttons&viewMode=story

Reply:
hmm. This seems like it should not be virtualizer specific. Don't we handle this somewhere else? 🤔

Reply:
We do handle some of this in [useGridState](https://github.com/adobe/react-spectrum/blob/a62195683addf229a536beabe95896c5e44d0dc9/packages/%40react-stately/grid/src/useGridState.ts#L55-L57)/[useListState](https://github.com/adobe/react-spectrum/blob/a62195683addf229a536beabe95896c5e44d0dc9/packages/%40react-stately/list/src/useListState.ts#L59-L61) but that only handles updating the tracked `focusedKey`